tensorboard/plugins/hparams/tf_hparams_query_pane/tf-hparams-query-pane.html

```diff
@@ -617,17 +635,36 @@
         const experimentRequest = {
           experimentName: this.experimentName,
         };
-        this.backend.getExperiment(experimentRequest).then((experiment) => {
-          if (_.isEqual(experiment, this._experiment)) {
-            // No need to update anything if there are no changes.
-            return;
-          }
-          this.set('_experiment', experiment);
-          this._computeHParams();
-          this._computeMetrics();
-          this._queryServer();
-          this._resolveGetExperiment();
-        });
+        this.backend
+          .getExperiment(experimentRequest)
+          .then((experiment) => {
+            if (_.isEqual(experiment, this._experiment)) {
+              // No need to update anything if there are no changes.
+              return;
+            }
+            this.set('_experiment', experiment);
+            // Doing this immediately, not only in finally(), allows the
+            // parent template to start rendering the results table before the
+            // table contents have been fetched.
+            this._computeDataFound();
+            this._computeHParams();
+            this._computeMetrics();
+            this._queryServer();
+            this._resolveGetExperiment();
```

Comment:
I don't fully understand why `computeDataFound` needs to happen before making the requests.  Doesn't the `finally` get called in the same task as the `then`?  Looking at devtools, the `finally` code executes before the network request for /session_groups completes.

Reply:
Oh!  Yes, great, thanks.